cmd/syncProcessor.go

```
@@ -334,11 +338,18 @@ func (b *remoteResourceDeleter) startDeleteWorkers(ctx context.Context) {
 	fmt.Printf("Waiting for deleteWorkers to return\n")
 	wg.Wait()
 	fmt.Printf("deleteWorkers Done\n")
+
+	// Flag "all deleted workers exited" to finalize(). It'll proceed on receiving this signal.
 	close(b.enumerationDone)
 }
 
 // deleteFolder list the files and add them to deleteChan for deletion.
 func (b *remoteResourceDeleter) deleteFolder(dirPath string) error {
+	// sanity check on dirPath.
+	if dirPath == "" {
+		return fmt.Errorf("cmd::deleteFolder called with empty directory path.")
```

let's panic on this, like we do elsewhere on unexpected condition.